org_fedora_oscap/model.py

```diff
@@ -0,0 +1,281 @@
+        finally:
+            with self.activity_lock:
+                self.now_fetching_or_processing = False
+
+        after_fetch(content)
+
+        return content
+
+    def _verify_fingerprint(self, dest_filename, fingerprint=""):
+        if not fingerprint:
+            return
+
+        hash_obj = utils.get_hashing_algorithm(fingerprint)
+        digest = utils.get_file_fingerprint(dest_filename,
+                                            hash_obj)
+        if digest != fingerprint:
+            log.error(
+                "File {dest_filename} failed integrity check - assumed a "
+                "{hash_obj.name} hash and '{fingerprint}', got '{digest}'"
+            )
+            msg = f"Integrity check of the content failed - {hash_obj.name} hash didn't match"
+            raise content_handling.ContentCheckError(msg)
+
+    def _finish_actual_fetch(self, wait_for, fingerprint, report_callback, dest_filename):
+        threadMgr.wait(wait_for)
+        actually_fetched_content = wait_for is not None
+
+        self._verify_fingerprint(dest_filename, fingerprint)
+
+        content = ObtainedContent(self.CONTENT_DOWNLOAD_LOCATION)
+
+        report_callback("Analyzing content.")
+        if not actually_fetched_content:
+            if not dest_filename:  # using scap-security-guide
+                fpaths = [f"{common.SSG_DIR}/{common.SSG_CONTENT}"]
+                labelled_files = content_handling.identify_files(fpaths)
+            else:  # Using downloaded XCCDF/OVAL/DS/tailoring
+                fpaths = glob(str(self.CONTENT_DOWNLOAD_LOCATION / "*.xml"))
+                labelled_files = content_handling.identify_files(fpaths)
+        else:
+            dest_filename = pathlib.Path(dest_filename)
+            # RPM is an archive at this phase
+            content_type = self.get_content_type(str(dest_filename))
+            if content_type in ("archive", "rpm"):
+                # extract the content
+                content.add_content_archive(dest_filename)
+                try:
+                    fpaths = common.extract_data(
+                        str(dest_filename),
+                        str(dest_filename.parent)
+                    )
+                except common.ExtractionError as err:
+                    msg = f"Failed to extract the '{dest_filename}' archive: {str(err)}"
+                    log.error(msg)
+                    raise err
+
+                # and populate missing fields
+                labelled_files = content_handling.identify_files(fpaths)
+
+            elif content_type == "file":
+                labelled_files = content_handling.identify_files([str(dest_filename)])
+            else:
+                raise common.OSCAPaddonError("Unsupported content type")
+
+        for f, l in labelled_files.items():
+            content.add_file(f, l)
+
+        if fingerprint:
+            content.record_verification(dest_filename)
+
+        return content
+
+
+class ObtainedContent:
```

Comment:
I feel that it somehow overlaps with OSCAPData class (forms ks/oscap.py).

Reply:
This one aims to assist the gathered files discovery - files can be downloaded directly, or they can be extracted. This class enables a user to quickly understand what is available, and whether the current set of contents is usable for further processing.